uuid.go

```diff
@@ -74,44 +78,58 @@ func Parse(s string) (UUID, error) {
 		}
 		s = s[9:]
 
+		// Ensure proper dash placement
+		if s[8] != '-' || s[13] != '-' || s[18] != '-' || s[23] != '-' {
+			return uuid, errors.New("invalid UUID format")
+		}
+		s = s[1:]
+
 	// {xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx}
 	case 36 + 2:
+		// Ensure proper dash placement
+		if s[8] != '-' || s[13] != '-' || s[18] != '-' || s[23] != '-' {
+			return uuid, errors.New("invalid UUID format")
+		}
 		s = s[1:]
 
 	// xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
 	case 32:
-		var ok bool
+		// Ensure proper length and valid characters
 		for i := range uuid {
-			uuid[i], ok = xtob(s[i*2], s[i*2+1])
-			if !ok {
+			uuid[i] = s[i]
```

Comment:
This code ignores half of the UUID and will provide the wrong answer.

It is possible that the existing for loop could be rewritten as 
```
for i := 0; i < 32; i += 2 {
    uuid[i>>1], ok = xtob(s[i], s[i+1])
    ...
```